src/charm.py

```diff
@@ -83,16 +83,20 @@ def __init__(self, *args):
         # self._profiling = ProfilingEndpointProvider(
         #     self, jobs=[{"static_configs": [{"targets": ["*:4080"]}]}]
         # )
-        # TODO:
-        #  ingress route provisioning a separate TCP ingress for each receiver if GRPC doesn't work directly
 
-        self._ingress = IngressPerAppRequirer(
-            self, port=self.tempo.tempo_server_port, strip_prefix=True
+        self._ingress = TraefikRouteRequirer(self, self.model.get_relation("ingress"), "ingress")  # type: ignore
+        self._tracing = TracingEndpointProvider(
+            self, host=self.tempo.host, external_url=self._ingress.external_host
```

Comment:
we're introducing an inconsistency here: 
- external url will be http or https depending on whether traefik has TLS active
- host will be a raw hostname, and the tracing lib will by default add http in front of it.

How about: 
- we rename host to internal_url (with a backwards-compat deprecation warning to give us time to adjust all dependent charms)
- we generate the internal_url from the tempo charm, not the charm lib, so the charm remains responsible of determining whether we should be using https or http. 

Other alternative would be to add to the lib an `internal_scheme` arg that the charm lib could use instead of the hardcoded `http`.

Reply:
True - I liked your first idea (host rename) a bit better, but I feel it's causing a bit more backwards compatibility issues than adding `internal_scheme` to the databag, so I went with the second option. There's still fallback to `http://` in case newer charm uses an older Tempo to send traces to as well.